### PR TITLE
#1843 Enables globalAssertSoftly through system properties

### PR DESCRIPTION
**Pull request by ashishkujoy**

### CODE REVIEW

File: kotest-framework/kotest-framework-api/src/commonMain/kotlin/io/kotest/core/internal/KotestEngineSystemProperties.kt [hunk 1]
```diff
@@ -63,4 +63,9 @@ object KotestEngineSystemProperties {
    *  If set -> filter testCases by this severity level and higher, else running all
    */
    const val severityPrefix = "kotest.framework.test.severity"
+
+   /**
+    * Enable assert softly globally.
+    * */
+   const val globalAssertSoftly = "kotest.framework.assertion.globalAssertSoftly"
 }
```
Comment (sksamuel) [inline kotest-framework/kotest-framework-api/src/commonMain/kotlin/io/kotest/core/internal/KotestEngineSystemProperties.kt:71]: Can we keep it all lower case

**Review by sksamuel — CHANGES_REQUESTED** [review]

File: kotest-framework/kotest-framework-api/src/commonMain/kotlin/io/kotest/core/internal/KotestEngineSystemProperties.kt (hunk 1 above)
Comment (ashishkujoy) [reply kotest-framework/kotest-framework-api/src/commonMain/kotlin/io/kotest/core/internal/KotestEngineSystemProperties.kt:71]: sure,

Comment (ashishkujoy) [reply kotest-framework/kotest-framework-api/src/commonMain/kotlin/io/kotest/core/internal/KotestEngineSystemProperties.kt:71]: done

**Comment by sksamuel** [comment]
Thanks @ashishkujoy. Can we update docs too.

**Pull request merged**